addons/account/tests/test_multivat.py

```diff
@@ -0,0 +1,230 @@
+class TestMultiVAT(TransactionCase):
+
+    @classmethod
+    @patch.object(AccountChartTemplate, '_get_chart_template_mapping', _get_chart_template_mapping)
+    def setUpClass(cls):
+        """
+            Setups a company with a custom chart template, containing a tax and a fiscal position.
+            We need to add xml_ids to the templates because they are loaded from their xml_ids
+        """
+        super().setUpClass()
+        instantiate_accountman(cls)
+
+        cls.company_1 = cls.env['res.company'].create({
+            'name': 'TestCompany1',
+            'country_id': cls.env.ref('base.be').id,
+        })
+
+        cls.user.write({
+            'company_ids': [Command.set(cls.company_1.ids)],
+            'company_id': cls.company_1.id,
+        })
+
+        test_get_data = data_method_provider("local", "be")
+        with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=test_get_data, autospec=True):
+            cls.env['account.chart.template'].try_loading('local', company=cls.company_1, install_demo=False)
+
+        foreign_country = cls.env.ref("base.fr")
+        cls.foreign_vat_fpos = cls.env["account.fiscal.position"].create({
+            "name": "FR foreign VAT",
+            "auto_apply": True,
+            "country_id": foreign_country.id,
+            "foreign_vat": "FR23334175221",
+        })
+
+        test_get_data = data_method_provider("foreign", "fr")
+        with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=test_get_data, autospec=True):
+            cls.foreign_vat_fpos.action_create_foreign_taxes()
+
+    def test_tax_and_tax_group_should_be_reachable_through_both_api(self):
+        # Ensure local and foreign tax xmld_id follow that pattern and that it's reachable using both api
+        for ref, xml_id in (
+                # tax group
+                (self.env.ref, f'account.{self.env.company.id}_tax_group_taxes'),
+                (self.env.ref, f'account.{self.env.company.id}_foreign_tax_group_taxes'),
+                (self.env["account.chart.template"].ref, 'tax_group_taxes'),
+                (self.env["account.chart.template"].ref, 'foreign_tax_group_taxes'),
+                # tax
+                (self.env.ref, f'account.{self.env.company.id}_test_tax_1_template'),
+                (self.env.ref, f'account.{self.env.company.id}_test_tax_2_template'),
+                (self.env.ref, f'account.{self.env.company.id}_foreign_test_tax_1_template'),
+                (self.env.ref, f'account.{self.env.company.id}_foreign_test_tax_2_template'),
+                (self.env["account.chart.template"].ref, 'test_tax_1_template'),
+                (self.env["account.chart.template"].ref, 'test_tax_2_template'),
+                (self.env["account.chart.template"].ref, 'foreign_test_tax_1_template'),
+                (self.env["account.chart.template"].ref, 'foreign_test_tax_2_template'),
```

Comment:
Can always use `self.env["account.chart.template"].ref`, it defaults to `self.env.ref` if there is a `.` in the ref